### ISSUE TITLE
colorama conflicts with pytest

### ISSUE THREAD

**Issue by wence-** [tkarna/cofs#1]
When running some of the tests I get the following error:

```
py.test test/operations/ 
============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.9, pytest-2.8.2, py-1.4.30, pluggy-0.3.1
rootdir: /data/lmitche1/models/cofs, inifile: 
plugins: pep8-1.0.6, ipdb-0.1.dev2, xdist-1.11, benchmark-2.4.1, cov-1.8.1
collected 32 items 

test/operations/test_operations_2d-3d.py ................................

================================================================================== 32 passed, 4 pytest-warnings in 2.81 seconds ==================================================================================
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/initialise.py", line 19, in reset_all
    AnsiToWin32(orig_stdout).reset_all()
  File "/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/ansitowin32.py", line 67, in __init__
    strip = conversion_supported or not is_a_tty(wrapped)
  File "/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/ansitowin32.py", line 17, in is_a_tty
    return hasattr(stream, 'isatty') and stream.isatty()
ValueError: I/O operation on closed file
Error in sys.exitfunc:
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/usr/lib/python2.7/atexit.pyc in _run_exitfuncs()
     22         func, targs, kargs = _exithandlers.pop()
     23         try:
---> 24             func(*targs, **kargs)
     25         except SystemExit:
     26             exc_info = sys.exc_info()

/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/initialise.pyc in reset_all()
     17 
     18 def reset_all():
---> 19     AnsiToWin32(orig_stdout).reset_all()
     20 
     21 

/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/ansitowin32.pyc in __init__(self, wrapped, convert, strip, autoreset)
     65         # should we strip ANSI sequences from our output?
     66         if strip is None:
---> 67             strip = conversion_supported or not is_a_tty(wrapped)
     68         self.strip = strip
     69 

/homes/lmitche1/.local/lib/python2.7/site-packages/colorama/ansitowin32.pyc in is_a_tty(stream)
     15 
     16 def is_a_tty(stream):
---> 17     return hasattr(stream, 'isatty') and stream.isatty()
     18 
     19 

ValueError: I/O operation on closed file
```

This appears to be because in this mode pytest has already closed stdout before colorama gets to it in atexit.  colorama.init is claimed to be a no-op on all platforms but windows.  But this is not the case, because it registers an atexit handler in all cases.  I think you're only using colorama for symbolic ansi escape sequences.  So I think one can get around this problem by just not calling colorama.init.  We're unlikely to run on windows any time soon.

**Comment by tkarna** [comment]
OK, good to know. We could detect if one is running pytest (or if stdout is colorama incompatible) and skip colorama.init(). If colorama causes too many issues I think we could remove it, it's just there for pretty printing.

**Comment by wence-** [comment]
I did that in #2.  So if that is merged, this issue can be closed.

**Comment by wence-** [comment]
Resolved by merging #2.
